src/pages/examSetter/Grade.tsx

```
@@ -0,0 +1,318 @@
+  if (feedback) {
+    correctPatterns = feedback.correct_points.map((point) => new RegExp(`\\b${point}\\b`, "gi"));
+    incorrectPatterns = feedback.incorrect_points.map((point) => new RegExp(`\\b${point}\\b`, "gi"));
+  }
+
+
+  // Avoid overlapping matches by tracking used ranges
+  const matchedRanges: { start: number; end: number; style: "correct" | "incorrect" }[] = [];
+
+  correctPatterns.forEach((pattern) => {
+    let match;
+    while ((match = pattern.exec(text)) !== null) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)